pkg/resourceinterpreter/interpreter.go

```diff
@@ -57,25 +58,29 @@ func NewResourceInterpreter(informer genericmanager.SingleClusterInformerManager
 }
 
 type customResourceInterpreterImpl struct {
-	informer genericmanager.SingleClusterInformerManager
-
-	customizedInterpreter   *customizedinterpreter.CustomizedInterpreter
-	defaultInterpreter      *defaultinterpreter.DefaultInterpreter
-	configurableInterpreter *configurableinterpreter.ConfigurableInterpreter
+	informer     genericmanager.SingleClusterInformerManager
+	interpreters []framework.Interpreter
 }
 
 // Start starts running the component and will never stop running until the context is closed or an error occurs.
 func (i *customResourceInterpreterImpl) Start(ctx context.Context) (err error) {
-	klog.Infof("Starting custom resource interpreter.")
-
-	i.customizedInterpreter, err = customizedinterpreter.NewCustomizedInterpreter(i.informer)
-	if err != nil {
-		return
+	klog.Infof("Starting resource interpreter.")
+	var interpreter framework.Interpreter
+	interpreterFactories := []func(genericmanager.SingleClusterInformerManager) (framework.Interpreter, error){
+		func(manager genericmanager.SingleClusterInformerManager) (framework.Interpreter, error) {
+			interpreter, err = configurableinterpreter.NewConfigurableInterpreter(manager)
+			return interpreter, err
+		},
```

Comment:
Doesn't the `ConfigurableInterpreter` implement the `framework.Interpreter`?
I don't get why we need to transform it here.